engine/src/main/java/org/hibernate/search/engine/search/predicate/dsl/RangePredicateMatchingStep.java

```diff
@@ -225,7 +227,50 @@ default N range(Range<?> range) {
 	 * before Hibernate Search attempts to interpret them as a field value.
 	 * See {@link ValueConvert} for more information.
 	 * @return The next step.
+	 * @deprecated Use {@link #within(Range, ValueConvert)} instead.
 	 */
-	N range(Range<?> range, ValueConvert convert);
+	@Deprecated
+	default N range(Range<?> range, ValueConvert convert) {
+		return within( range, convert );
+	}
+
+	/**
+	 * Require at least one of the targeted fields to be in the given range.
+	 *
+	 * @param range The range to match.
+	 * The signature of this method defines this parameter as a range with bounds of any type,
+	 * but a specific type is expected depending on the targeted field.
+	 * See {@link ValueConvert#YES} for more information.
+	 * @return The next step.
+	 */
+	default N within(Range<?> range) {
+		return within( range, ValueConvert.YES );
+	}
+
+	/**
+	 * Require at least one of the targeted fields to be in the given range.
+	 *
+	 * @param range The range to match.
+	 * The signature of this method defines this parameter as a range with bounds of any type,
+	 * but a specific type is expected depending on the targeted field and on the {@code convert} parameter.
+	 * See {@link ValueConvert} for more information.
+	 * @param convert Controls how the range bounds should be converted
+	 * before Hibernate Search attempts to interpret them as a field value.
+	 * See {@link ValueConvert} for more information.
+	 * @return The next step.
+	 */
+	N within(Range<?> range, ValueConvert convert);
+
+	/**
+	 * Require at least one of the targeted fields to be in the given range.
+	 *
+	 * @param ranges The ranges to match.
+	 * The signature of this method defines this parameter as a range with bounds of any type,
+	 * but a specific type is expected depending on the targeted field.
+	 * See {@link ValueConvert#YES} for more information.
+	 * @return The next step.
+	 */
+	N withinAny(Range<?>... ranges);
 
+	RangePredicateMatchingWithinAnyMoreStep<?> withinAny();
```

Comment:
Unless you already found similar syntax elsewhere, I would recommend sticking to `withinAny(Range<?>...)` / `withinAny(Collection<? extends Range<?>>)` / `withinAny(Collection<? extends Range<?>>, ValueConvert)`.

See `org.hibernate.search.engine.search.predicate.dsl.TermsPredicateMatchingStep`.

If people want to use a different `ValueConvert` for each range, let them find a solution for themselves. There are limits to how flexible we should be, IMO.

Reply:
Thanks for taking a look at this one! 😃 👍🏻 ok will do😃